### PR TITLE
Improve offline deals

### PR DESCRIPTION
**Pull request by arajasek**
With these changes, the offline deal workflow succeeds

### CODE REVIEW

File: storagemarket/types.go [hunk 1]
```diff
@@ -426,6 +426,9 @@ type StorageClient interface {
 	// ProposeStorageDeal initiates deal negotiation with a Storage Provider
 	ProposeStorageDeal(ctx context.Context, addr address.Address, info *StorageProviderInfo, data *DataRef, startEpoch abi.ChainEpoch, endEpoch abi.ChainEpoch, price abi.TokenAmount, collateral abi.TokenAmount, rt abi.RegisteredProof) (*ProposeStorageDealResult, error)
 
+	// CalculateCommP calculates the piece CID for a given CAR file and miner
```
Comment (whyrusleeping) [inline storagemarket/types.go:429]: comment seems slightly inaccurate

File: storagemarket/impl/client.go [hunk 2]
```diff
@@ -199,9 +200,21 @@ func (c *Client) ProposeStorageDeal(
...
 ) (*storagemarket.ProposeStorageDealResult, error) {
-	commP, pieceSize, err := clientutils.CommP(ctx, c.pio, rt, data)
-	if err != nil {
-		return nil, xerrors.Errorf("computing commP failed: %w", err)
+
+	var commP cid.Cid
+	var pieceSize abi.UnpaddedPieceSize
+	var err error
+
+	if data.TransferType != storagemarket.TTManual {
```
Comment (whyrusleeping) [inline storagemarket/impl/client.go:208]: the clientutils.CommP function already checks if its a manual transfer and short circuits. Maybe move this logic down into that?

Comment (hannahhoward) [reply storagemarket/impl/client.go:208]: yea that makes sense

**Comment by arajasek** [comment]
Thanks for the suggestions, @hannahhoward, I didn't realise after some edits that I wound up just calling a public static method! I've switched to doing it directly from Lotus, I think it's nicer :) 

All suggested changes implemented :)

**Pull request merged**